idaes/core/base/unit_model.py

```diff
@@ -166,11 +166,13 @@ def add_port(self, name=None, block=None, doc=None):
         """
         # Create Port
         try:
-            p = block.build_port(self, name, doc)
+            # FIXME: What if name is None here? We probably need a default.
+            p = block.build_port_and_add(self, name, doc)
```

Comment:
If name is None we get an error in `build_port_and_references`, right? My instinct would be to add:
```python
if name is None:
    name = "port"
```